fga/client/client.go

```diff
@@ -0,0 +1,30 @@
+package client
+
+import (
+	"time"
+
+	"github.com/jellydator/ttlcache/v3"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openmfp/golang-commons/fga"
+)
+
+var _ fga.OpenFGAClientServicer = (*OpenFGAClient)(nil)
+
+type OpenFGAClient struct {
+	client openfgav1.OpenFGAServiceClient
+	cache  *ttlcache.Cache[string, string]
+}
+
+func NewOpenFGAClient(openFGAServiceClient openfgav1.OpenFGAServiceClient) (*OpenFGAClient, error) {
```

Comment:
I moved grpc connection initiation outside the constructor to make this code testable. Now we deal with the interface which can be mocked.
Old code to compare is [here](https://github.tools.sap/dxp/golang-dxp/blob/main/fga/client/client.go#L35-L54)